File: Python/asteroid_trajectory_selection/score_paths_mars.py
from .unpack_input import unpack_mars_input
from .compute_mars_path_deltav import compute_mars_path_deltav


def score_paths_mars(input_vec, a_id_1, a_id_2, a_id_3, launch_range,
                     m_1, m_2, m_3, m_mars):
    et_launch, et_mars, et_arrive_1, et_stay_1, et_arrive_2, et_stay_2, et_arrive_3 = \
        unpack_mars_input(input_vec, launch_range)

    result = compute_mars_path_deltav(a_id_1, a_id_2, a_id_3, et_launch, et_mars,
                                      et_arrive_1, et_stay_1, et_arrive_2,
                                      et_stay_2, et_arrive_3, m_1, m_2, m_3, m_mars)

    return result['delta_v_total']

File: Trajectory Selection/Code/Asteroid_Trajectory_Selection/SCORE_PATHS_MARS.m
function return_score = SCORE_PATHS_MARS(input_vec, A_ID_1, A_ID_2, A_ID_3, ...
    LAUNCH_RANGE, M_1, M_2, M_3, M_MARS)

    [et_launch, et_mars, et_arrive_1, et_stay_1, et_arrive_2, ...
        et_stay_2, et_arrive_3] = UNPACK_MARS_INPUT(input_vec, LAUNCH_RANGE);



    [~, ~, ~, ~, ~, ~, ~, ~, delta_v_total] = COMPUTE_MARS_PATH_DELTAV(A_ID_1, ...
        A_ID_2, A_ID_3, et_launch, et_mars, et_arrive_1, et_stay_1, et_arrive_2, ...
        et_stay_2, et_arrive_3, M_1, M_2, M_3, M_MARS);
    
    return_score = delta_v_total;
end


